src/lib/build.ts

```diff
@@ -58,84 +51,31 @@ export default async function run({
       : outputPath;
   const outputSize = parseSize(size);
 
-  try {
-    const [source, broker] = await launchSourceAndBrokerServer(root);
-    const sourcePort = source.port;
-    const brokerPort = broker.port;
-    const navigateURL = getBrokerUrl({
-      sourcePort,
-      sourceIndex,
-      brokerPort,
-      loadMode,
-      outputSize,
-    });
-    const chromePort = await findPort();
-    const launcherOptions = {
-      port: chromePort,
-      chromeFlags: [
-        '--disable-gpu',
-        '--headless',
-        sandbox ? '' : '--no-sandbox',
-      ],
-    };
-
-    console.log(`Launching headless Chrome... port:${launcherOptions.port}`);
-
-    await launchChrome(launcherOptions).catch((err) => {
-      if (err.code === 'ECONNREFUSED') {
-        console.log(`Cannot launch headless Chrome. use --no-sandbox option.`);
-      } else {
-        console.log(
-          'Cannot launch Chrome. Did you install it?\nvivliostyle-cli supports Chrome (Canary) only.',
-        );
-      }
-      process.exit(1);
-    });
+  const [source, broker] = await launchSourceAndBrokerServer(root);
+  const sourcePort = source.port;
+  const brokerPort = broker.port;
+  const navigateURL = getBrokerUrl({
+    sourcePort,
+    sourceIndex,
+    brokerPort,
+    loadMode,
+    outputSize,
+  });
 
-    chrome({ port: chromePort }, async (protocol) => {
-      const { Page, Runtime, Emulation } = protocol;
-
-      await Promise.all([Page.enable(), Runtime.enable()]).catch((err) => {
-        console.trace(err);
-        process.exit(1);
-      });
-
-      Page.loadEventFired(async () => {
-        await onPageLoad({
-          Page,
-          Runtime,
-          Emulation,
-          outputFile,
-          vivliostyleTimeout,
-        }).catch((err: Error) => {
-          console.trace(err);
-          protocol.close();
-          process.exit(1);
-        });
-
-        protocol.close();
-        process.exit(0);
-      });
-
-      Page.navigate({ url: navigateURL });
-    }).on('error', (err) => {
-      console.error('Cannot connect to Chrome:' + err);
-      process.exit(1);
-    });
-  } catch (err) {
-    console.trace(err);
-    process.exit(1);
-  }
-}
+  log(`Launching build environment... `);
+  const browser = await puppeteer.launch({
+    headless: true,
+    args: ['--disable-gpu', sandbox ? '' : '--no-sandbox'],
```

Comment:
`--disable-gpu` shouldn't be necessary for Docker/Linux to work in headless mode for the more recent Chromium versions included with Puppeteer. It appears to speed up printing by a couple of seconds on macOS, too, as far as I can tell. 😺

Reply:
I have no idea what the purpose of `--disable-gpu` here though if it's clear that it would work in macOS / Linux / Windows without this flag, I will omit it.
cc: @spring-raining

Reply:
Puppeteer seems to add appropriate launch arguments as Zwaldowski says. Let's omit `--disable-gpu`! https://github.com/puppeteer/puppeteer/pull/2908

Reply:
done! https://github.com/vivliostyle/vivliostyle-cli/pull/32/commits/0fd7b6a2251197accc2257a49085bd3824a4f358